implement-directus-auth-in-nextjs-14/index.md

```diff
@@ -0,0 +1,995 @@
+---
+title: 'Implement Directus Auth in Next.js 14'
+description: 'A tutorial on implementing Authentication on a Next.js blog website using Directus as a Headless CMS.'
+author:
+  name: 'Emmanuel John'
+  avatar_file_name: 'profile-photo.jpeg'
+---
+
+## Introduction
+In this tutorial, you will learn how to implement Authentication on a Next.js blog website using Directus as a Headless CMS. You’ll register new users, login, handle refresh tokens, create posts, list all posts, and edit/delete posts where the auth user is the post creator.
+
+## Before You Start
+To follow along with this tutorial, you will need the following knowledge and tools:
+
+- [Node.js](https://nodejs.org/en/download/) v18 and above installed on your computer.
+- A Directus project - follow our [quickstart guide](https://docs.directus.io/blog/docs.directus.io/getting-started/quickstart.html) if you don't already have one.
+- Prior experience with Next.js.
+
+<!-- ## Your Sections Here -->
+## Allowing Public Role To Create Users
+
+1. Navigate to `Settings`.
+2. Select `Access Control`.
+3. Click on the `Public Role`.
+4. Access the `System Collection`.
+5. Locate `directus_users`.
+6. Under the `Create` option, choose `Use Custom`.
+7. Enable field permissions for `first_name`, `last_name`, `role`, `Email` and `Password` options.
+![Directus public role field permission settings](public-role-field-permission-settings.png)
+8. Enable field validation where `role` equals the Author role `Primary Key`
+![Directus public role field validation settings](public-role-field-validation-settings.png)
```

Comment:
```suggestion
## Allowing Public Registration

When a user first registers, they will be unauthenticated. In Directus, the permissions of unauthenticated users are controlled by the public role.

In your access control settings open the public role. Find `directus_users` under system collections, and then set custom permissions.

1. In field permissions, enable only `first_name`, `last_name`, `email` and `password` options. If you want users to provide other data at the time of registration, enable the respective field.
2. In field validation, require `first_name`, `last_name`, `email`, and `password` to not be empty.
3. In field presets, set the `role` to the `id` of the author role.

This combination of settings in a custom permission will ensure users provide the required fields, and automatically set the role.
```

Your approach is valid, but it's safer to not expose the role setting to the client. By not allowing the field to be set by the public role, and using a field preset, new users are given the role automatically. 

This section also has to go after the creation of the author role, as it's referenced here.